sdk/storage/storage-blob/review/storage-blob.api.md

```diff
@@ -1664,6 +1683,22 @@ export interface ContainerExistsOptions extends CommonOptions {
     abortSignal?: AbortSignalLike;
 }
 
+// @public
+export interface ContainerGenerateSasUrlOptions {
```

Comment:
there seems to be a lot of overlap with `BlobGenerateSasUrlOptions` - should we have a common base interface for SasUrlOptions ? Or at least have these two extend `ServiceGenerateAccountSasUrlOptions ` ?

Reply:
I thought about using the same interface for `BlobGenerateSasUrlOptions` and `ContainerGenerateSasUrlOptions` as they only differ in the type for `permissions`. And we are using the same interface for Blob and Container when generting the sas in `BlobSASSignatureValues`.

I  am fine with having `ContainerGenerateSasUrlOptions` and `BlobGenerateSasUrlOptions` to extend a common base interface or simply use a same interface for now. But I don't want to the base to be `ServiceGenerateAccountSasUrlOptions` since the account sas generation is different with that of Blob and Container. For example, it doesn't support stored access policy so required fields like `expiresOn` and `permissions` are pulled out from the option. I am concerned that there may be other differences in the future. And only three interfaces have these common fields so it looks like an overkill to define a common base interface for it.

Reply:
In datalake, the FileSystem, Directory and File share the same `DataLakeSASSignatureValues` to generate sas but we have to add `isDirectory` and additional comments to differentiate.

I am suggesting share a common base interface for service sas generate options, i.e. Container and Blob; FileSystem, Directory and File. And the account sas option should be on its own.